File: frontend/src/api/training.ts
import http from './index'

export function createTrainingApi(data: {
  student_id: number
  week_number: number
  year: number
  technique_score: number
  fitness_score: number
  match_score: number
  notes?: string
}) {
  return http.post('/training', data)
}

export function getTrainingByStudentApi(studentId: number, params?: { year?: number }) {
  return http.get(`/training/student/${studentId}`, { params })
}

export function getTrainingApi(id: number) {
  return http.get(`/training/${id}`)
}

File: backend/app/models/training.py
from datetime import datetime

from sqlalchemy import String, Integer, Float, Text, DateTime, ForeignKey, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class TrainingRecord(Base):
    __tablename__ = "training_records"
    __table_args__ = (
        UniqueConstraint("student_id", "week_number", "year", name="uq_training_student_week_year"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    student_id: Mapped[int] = mapped_column(ForeignKey("students.id"), nullable=False)
    coach_id: Mapped[int] = mapped_column(ForeignKey("coaches.id"), nullable=False)
    week_number: Mapped[int] = mapped_column(Integer, nullable=False)
    year: Mapped[int] = mapped_column(Integer, nullable=False)
    technique_score: Mapped[float] = mapped_column(Float, default=0.0)
    fitness_score: Mapped[float] = mapped_column(Float, default=0.0)
    match_score: Mapped[float] = mapped_column(Float, default=0.0)
    notes: Mapped[str] = mapped_column(Text, default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), onupdate=func.now())
